src/lib/local-storage-utils.js

```diff
@@ -45,3 +45,45 @@ export const getOrCreateValidCartId = () => {
 
     return storedValue;
 };
+
+/* Sets a new userId to expire in one month */
+export const setUserId = (userId : string) => {
+    const storedValue = {
+        userId,
+        createdAt: Date.now()
+    };
+
+    window.localStorage.setItem(storage.paypalCrUser, JSON.stringify(storedValue));
+
+    return storedValue;
+};
+
+/* Generates a random userId to expire in one month */
+export const createNewUserId = () => {
+    const userId = `${ generate.generateId() }`;
+
+    return setUserId(userId);
+};
+
+/* Returns a userId if one exists */
+export const getUserId = () => {
+    const storedValue = window.localStorage.getItem(storage.paypalCrUser);
+
+    if (storedValue) {
+        return JSON.parse(storedValue);
+    }
+
+    return null;
+};
+
+/* Returns an existing, valid userId or creates a new one */
+export const getOrCreateValidUserId = () => {
```

Comment:
invalid = userId > 30 days.

Reply:
"invalid" would be expired. The extra word is there to indicate that it isn't necessarily a pure get-or-create, since a value might already exist but we consider it expired.